lib/clients/GcpClient.py

```diff
@@ -0,0 +1,555 @@
+                 poll_maximum_time):
+        super(GcpClient, self).__init__(operation_name, configuration, directory_persistent, directory_work_list,
+                                        poll_delay_time, poll_maximum_time)
+
+        self.__gcpCredentials = json.loads(configuration['credentials'])
+        self.project_id = configuration['projectId']
+        self.tags = {
+            'instance_id': self.INSTANCE_ID,
+            'job_name': self.JOB_NAME
+        }
+
+        self.compute_api_name = 'compute'
+        self.compute_api_version = 'v1'
+        self.snapshot_prefix = 'sf-snapshot'
+        self.disk_prefix = 'sf-disk'
+        self.device_path_template = '/dev/disk/by-id/google-{}'
+
+        self.compute_client = self.create_compute_client()
+        self.storage_client = self.create_storage_client()
+
+        # +-> Check whether the given container exists and is accessible
+        self.container = self.get_container()
+        if not self.container:
+            msg = 'Could not find or access the given container.'
+            self.last_operation(msg, 'failed')
+            raise Exception(msg)
+
+        # +-> Get the availability zone of the instance
+        self.availability_zone = self._get_availability_zone_of_server(
+            configuration['instance_id'])
+        if not self.availability_zone:
+            msg = 'Could not retrieve the availability zone of the instance.'
+            self.last_operation(msg, 'failed')
+            raise Exception(msg)
+
+    def create_compute_client(self):
+        try:
+            credentials = service_account.Credentials.from_service_account_info(
+                self.__gcpCredentials)
+            compute_client = discovery.build(
+                self.compute_api_name, self.compute_api_version, credentials=credentials)
+            return compute_client
+        except Exception as error:
+            raise Exception(
+                'Creation of compute client failed: {}'.format(error))
+
+    def create_storage_client(self):
+        try:
+            credentials = service_account.Credentials.from_service_account_info(
+                self.__gcpCredentials)
+            storage_client = storage.Client(self.project_id, credentials)
+            return storage_client
+        except Exception as error:
+            raise Exception(
+                'Creation of storage client failed: {}'.format(error))
+
+    def _get_availability_zone_of_server(self, instance_id):
+        try:
+            expression = 'name eq ' + instance_id
+            request = self.compute_client.instances().aggregatedList(
+                project=self.project_id, filter=expression)
+            while request is not None:
+                response = request.execute()
+                for zone, instances_scoped_list in response['items'].items():
+                    if 'warning' in instances_scoped_list:
+                        pass
```

Comment:
Extra if check doing nothing. 
Can can have only one if check which checks the case below?

Reply:
Yes. will make the change.